libcontainer/configs/validate/validator.go

```diff
@@ -156,6 +169,19 @@ func checkHostNs(sysctlConfig string, path string) error {
 	if err != nil {
 		return fmt.Errorf("read soft link %q error", currentProcessNetns)
 	}
+
+	// First check if the provided path is a symbolic link
+	symLink, err := isSymbolicLink(path)
+	if err != nil {
+		return fmt.Errorf("could not check that %q is a symlink: %v", path, err)
+	}
+
+	if symLink == false {
+		// The provided namespace is not a symbolic link,
+		// it is not the host namespace.
+		return nil
+	}
+
 	// readlink on the path provided in the struct
 	destOfContainer, err := os.Readlink(path)
```

Comment:
@cyphar If `path` is not symlink, i.e. a persistent namespace, it will *not* error out.
It will only error out if `path` is a symlink and point to the same namespace as `/proc/self/ns/net`, which is considered to be the host namespace.

In other words, this routine will error out if `path` is not pointing at a persistent namespace and if it's a symbolic link pointing at the host namespace.